### PR TITLE
PLG-107: In product grid, improve the page title when a view is selected

### PR DESCRIPTION
**Pull request by BryanS29-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/grid/view-selector.js [hunk 1]
```diff
@@ -446,12 +460,14 @@ define([
     selectView: function (view) {
       DatagridState.set(this.gridAlias, {
         view: view.id,
+        text: view.text,
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/grid/view-selector.js:463]: As we do not use directly the `DatagridState` service to display the `text` parameter. We should remove this line.

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/common/product/ProductGridContext.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  children: string;
+};
+
+const ProductGridContext = ({type, children}: Props) => {
+  const translate = useTranslate();
+
+  return (
+    <ContextContainer>
+      <ViewNameContainer>{children}</ViewNameContainer>{' '}
+      {type === 'public' || type === 'view' ? `(${translate('pim_common.public_view')})` : null}
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/common/product/ProductGridContext.tsx:16]: In order to ease the readability of the code, we could manage the separating space directly in the template string like below:

```suggestion
      <ViewNameContainer>{children}</ViewNameContainer>
      {type === 'public' || type === 'view' ? ` (${translate('pim_common.public_view')})` : null}
```

**Comment by olivierpontier** [comment]
The `form extensions` allows to define multiple modules to display in a same `dropZone`. So, the `pim-product-index-grid-context` code seems to not be appropriate if we had to display others modules in the `context` drop zone.
Furthermore, we try to name our modules and components in a way to identify what they do.

We should rename the different codes, modules and components in something like `ProductGridViewTitle`. @BryanS29-akeneo @rouliane @LaurentPetard what do you think about it? Do we have suggestion? :pray:

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/common/product/ProductGridViewTitle.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  type: string;
+  projectDetails?: ProjectDetails;
+  children: string;
+};
+
+type ProjectDetails = {
+  dueDateLabel: string;
+  dueDate: string;
+  completionRatio: number;
+  badgeClass: string;
```
Comment (olivierpontier) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/common/product/ProductGridViewTitle.tsx:15]: As we using React component, and the Badge component is existing in the DSM, the badgeClass property does not seem to be useful

**Review by olivierpontier — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/common/product/product-grid-view-title-context.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import {ProductGridViewTitle} from '../../../grid/ProductGridViewTitle';
+
+const BaseView = require('pimui/js/view/base');
+const mediator = require('oro/mediator');
+
+type CurrentView = {
```
Comment (LaurentPetard) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/form/common/product/product-grid-view-title-context.ts:6]: As the type is contextual (not exported) I think that it's better to define only the attributes you need (so only `type` and `text`)

**Review by LaurentPetard — APPROVED** [review]

**Pull request merged**